server/storage/sqldb.go

```diff
@@ -176,42 +220,46 @@ func (db *SQLStorage) writeChangefeed(tx *gorm.DB, gun data.GUN, version int, ch
 }
 
 // GetCurrent gets a specific TUF record
-func (db *SQLStorage) GetCurrent(gun data.GUN, tufRole data.RoleName) (*time.Time, []byte, error) {
+func (db *SQLStorage) GetCurrent(gun data.GUN, tufRole data.RoleName, channels ...*Channel) (*time.Time, []byte, error) {
 	var row TUFFile
-	q := db.Select("updated_at, data").Where(
-		&TUFFile{Gun: gun.String(), Role: tufRole.String()}).Order("version desc").Limit(1).First(&row)
+	q := db.Select("tuf_files.updated_at, data").Scopes(TufFilesInChannels(channels...)).
+		Where(&TUFFile{
+			Gun:  gun.String(),
+			Role: tufRole.String(),
+		}).
+		Order("version desc").Limit(1).First(&row)
 	if err := isReadErr(q, row); err != nil {
 		return nil, nil, err
 	}
 	return &(row.UpdatedAt), row.Data, nil
 }
 
 // GetChecksum gets a specific TUF record by its hex checksum
-func (db *SQLStorage) GetChecksum(gun data.GUN, tufRole data.RoleName, checksum string) (*time.Time, []byte, error) {
+func (db *SQLStorage) GetChecksum(gun data.GUN, tufRole data.RoleName, checksum string, channels ...*Channel) (*time.Time, []byte, error) {
 	var row TUFFile
-	q := db.Select("created_at, data").Where(
-		&TUFFile{
+	q := db.Select("tuf_files.created_at, data").Scopes(TufFilesInChannels(channels...)).
+		Where(&TUFFile{
 			Gun:    gun.String(),
 			Role:   tufRole.String(),
 			SHA256: checksum,
-		},
-	).First(&row)
+		}).
+		Order("version desc").Limit(1).First(&row)
```

Comment:
I agree, I see no reason this is needed and can't remember why I changed it. It's potentially just a copy/paste error.